### PR TITLE
EU-82; Add Dockerfile and docker-compose file example

### PR DESCRIPTION
**Pull request by belkka**
Add requirements.txt with list of python dependencies.

Define Dockerfile (and .dockerignore) to build image and run application in docker container.

Add examples of docker-compose file and .env file that may be useful for local development.

Update README.md, describe how to run the app locally using docker-compose file.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.8
```
Comment (belkka) [inline Dockerfile:1]: Not sure which base image is better to use

**Comment by belkka** [comment]
Depends on https://github.com/anton-serozhechkin/e_university_api/pull/44 (Dockerfile now uses poetry).

File: local/docker-compose.example.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+version: '3'
+
+services:
+  database:
```
Comment (anton-serozhechkin) [inline local/docker-compose.example.yml:4]: please rename to, for eg postgres or postgresql

File: local/docker-compose.example.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    image: 'postgres:latest'
+    env_file:
+      - .env  # must define POSTGRES_USER, POSTGRES_PASSWORD and POSTGRES_DB
+    volumes:
+      - database-data:/var/lib/postgresql/data/  # persist data
+    ports:
+      - '5432:5432'  # required to access DB from host system
```
Comment (anton-serozhechkin) [inline local/docker-compose.example.yml:11]: please add container name

File: local/docker-compose.example.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    build: .
+    environment:
+      DATABASE_URL: "postgresql://${POSTGRES_USER}:${POSTGRES_PASSWORD}@database:5432/${POSTGRES_DB}"
+    env_file:
+      - .env
+    depends_on:
+      - database
+    command: uvicorn --host 0.0.0.0 --port 8000 main:app
+    ports:
+      - '8000:8000'
```
Comment (anton-serozhechkin) [inline local/docker-compose.example.yml:22]: please add container name

File: README.md [hunk 5]
```diff
@@ -1 +1,76 @@
...
+
+1. Install Python 3.8 (refer `pyproject.toml` for python verion used on the project) and [Poetry](https://python-poetry.org/)
+
+2. Create virtual environment and install project dependencies:
+
+       poetry install
+
+3. Run database in docker container using docker-compose:
+
+       docker-compose up database
```
Comment (anton-serozhechkin) [inline README.md:50]: don't forget to rename

File: README.md [hunk 6]
```diff
@@ -1 +1,76 @@
...
+
+During development it may be useful to run application outside of docker container (e. g. for debugging). To perform this:
+
+1. Install Python 3.8 (refer `pyproject.toml` for python verion used on the project) and [Poetry](https://python-poetry.org/)
+
+2. Create virtual environment and install project dependencies:
+
+       poetry install
+
+3. Run database in docker container using docker-compose:
```
Comment (anton-serozhechkin) [inline README.md:48]: don't forget to rename

File: README.md [hunk 7]
```diff
@@ -1 +1,76 @@
...
+
+       docker-compose up backend  # automatically runs database service as dependency
+
+   or (shorter)
+
+       docker-compose up  # run all services defined in docker-compose file
+
+   > :warning: If you see an error messages about failing to connect database, try running database *first*:
+   >
+   >     docker-compose up database  # wait several seconds until database is up
```
Comment (anton-serozhechkin) [inline README.md:32]: don't forget to rename

**Review by anton-serozhechkin — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (belkka) [reply README.md]: renamed service to "postgres"

File: local/docker-compose.example.yml (hunk 2 above)
Comment (belkka) [reply local/docker-compose.example.yml]: done

File: local/docker-compose.example.yml (hunk 4 above)
Comment (belkka) [reply local/docker-compose.example.yml:26]: added above

**Review by anton-serozhechkin — APPROVED** [review]

**Pull request merged**